### PR TITLE
refactor(v3/vc-selector/input): use composition api

### PR DESCRIPTION
**Pull request by buqiyuan**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch. Pull request will be merged after one of collaborators approve. Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [x] TypeScript definition update
- [x] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> 1. Describe the source of requirement.
> 2. Resolve what problem.
> 3. Related issue link.

### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
> 2. Chinese description (optional)

### Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

**Comment by buqiyuan** [comment]
image
为什么这里的a-textarea不能有自定义class

**Comment by tangjinzhou** [comment]
bug

File: components/vc-select/Selector/Input.tsx [hunk 1]
```diff
@@ -168,25 +149,24 @@ const Input = defineComponent({
...
-        inputNode.type === 'textarea' ? {} : { type: 'search' },
-      ),
-      true,
-      true,
-    ) as VNode;
-    return inputNode;
+        InputNode.type === 'textarea' ? {} : { type: 'search' },
+      );
+
+      return <InputNode {...inputProps}></InputNode>;
```
Comment (tangjinzhou) [inline components/vc-select/Selector/Input.tsx:168]: 这么写应该默认合并了属性的，会支持触发原inputNode 的事件，这里又做了onOriginXXX ，应该会触发两次
class 也是这个原因，导致 class 挂载上了(反而成了正确的结果)，问题的根本原因是  `InputNode?.props?.className` 是错的，应该是 `InputNode?.props?.class`

**Review by tangjinzhou — CHANGES_REQUESTED** [review]

File: components/vc-select/Selector/Input.tsx (hunk 1 above)
Comment (tangjinzhou) [reply components/vc-select/Selector/Input.tsx:168]: 保留 cloneElement 的逻辑，className 改成 class  好了

Comment (buqiyuan) [reply components/vc-select/Selector/Input.tsx:168]: ok

Comment (buqiyuan) [reply components/vc-select/Selector/Input.tsx]: done

**Comment by buqiyuan** [comment]
@tangjinzhou 对了，唐大佬，其实现在在业务中我对antdv的二次封装还遇到一些小问题，就是antdv导出的props中带有onXxx的事件函数的问题，会导致@ok这些事件会被vue合并成为一个数组，我觉得除了hook函数式使用的组件的props里的事件可以是onXxx，普通组件的props里不应当有onXxx，组件事件都应该是定义在emit中，这样的话 使用的时候volar也会有提示的，所以我现在遇到的小麻烦是如图所示，我需要把重复的事件omit掉：
image
不知道我描述的清楚不

**Comment by buqiyuan** [comment]
配上warn信息应该好理解：
image

**Pull request merged**